lib/improver/utilities/cube_units.py

```diff
@@ -93,28 +94,37 @@ def enforce_units_and_dtypes(cubes, coords=None, enforce=True):
             conforms = _check_units_and_dtype(item, units, dtype)
             if not conforms:
                 msg = ('{} with units {} and datatype {} does not conform'
-                       ' to expected standard (units {}, datatype {})')
+                       ' to expected standard (units {}, datatype {})\n')
                 msg = msg.format(item.name(), item.units, item.dtype,
                                  units, dtype)
-                raise ValueError(msg)
+                error_string += msg
+                continue
 
-        # convert units
+        # attempt to convert units and record any errors
         try:
             item.convert_units(units)
         except ValueError:
-            msg = '{} units cannot be converted to "{}"'
-            raise ValueError(msg.format(item.name(), units))
+            msg = '{} units cannot be converted to "{}"\n'
+            error_string += msg
 
-        # convert datatype
-        if isinstance(item, iris.cube.Cube):
-            _convert_diagnostic_dtype(item, dtype)
-        else:
-            _convert_coordinate_dtype(item, dtype)
+        # attempt to convert datatype and record any errors
+        try:
+            if isinstance(item, iris.cube.Cube):
+                _convert_diagnostic_dtype(item, dtype)
+            else:
+                _convert_coordinate_dtype(item, dtype)
+        except ValueError as cause:
+            error_string += cause + '\n'
+
+    # if any errors were raised, re-raise with all messages here
+    if error_string:
+        msg = 'The following errors were raised during processing:\n'
+        raise ValueError(msg+error_string)
```

Comment:
Thanks for consolidating all this, it might be helpful (though hopefully we won't be making multiple mistakes 🤞 ).